cloud_utils/vault.py

```diff
@@ -1,136 +1,51 @@
-from typing import Any, Callable, Coroutine, Dict, Optional
+from typing import Callable, Optional
 
-import gamla
-from cache import AsyncTTL
+import hvac
+from hvac import exceptions
+from hvac.api.auth_methods import Kubernetes
 
-base_metadata_url = "http://169.254.169.254/metadata"
+_MOUNT_POINT = "secret"
 
 
-class KeyDoesNotExist(Exception):
-    pass
-
-
-async def _identity_token():
-    return (
-        (
-            await gamla.get_async_with_headers(
-                {"Metadata": "true"},
-                30,
-                f"{base_metadata_url}/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fmanagement.azure.com%2F",
-            )
+def _build_read_all_secrets(client: hvac.Client):
+    def read_all_secrets(path: str, version: Optional[str]) -> dict[str, str]:
+        read_secrets = client.secrets.kv.v2.read_secret_version(
+            path=path, version=version
         )
-        .json()
-        .get("access_token")
-    )
+        return read_secrets["data"]["data"]
```

Comment:
What happens if the key does not exist? Do we want to catch and raise our own error?